pex/environment.py

```diff
@@ -111,21 +109,14 @@ def _force_local(cls, pex_file, pex_info):
       return pex_file
     explode_dir = os.path.join(pex_info.zip_unsafe_cache, pex_info.code_hash)
     TRACER.log('PEX is not zip safe, exploding to %s' % explode_dir)
-    if not os.path.exists(explode_dir):
-      explode_tmp = explode_dir + '.' + uuid.uuid4().hex
-      with TRACER.timed('Unzipping %s' % pex_file):
-        try:
-          safe_mkdir(explode_tmp)
+    with atomic_directory(explode_dir) as explode_tmp:
+      if explode_tmp:
+        with TRACER.timed('Unzipping %s' % pex_file):
           with open_zip(pex_file) as pex_zip:
             pex_files = (x for x in pex_zip.namelist()
```

Comment:
Is this logic likely to ever be used elsewhere? Could it perhaps be moved into a separate method named something like `_iter_non_cache_bootstrap_entries(pex_zip)`? "No" is a great answer -- just curious. This code is pretty readable though.

Reply:
Not likely, no.